source/renderer/zettlr-editor.js

```diff
@@ -222,6 +222,39 @@ class ZettlrEditor {
 
     this._searcher.setInstance(this._cm)
 
+    this._cm.on('keyup', (cm, event) => {
+      if (event.altKey &&
+        event.code === 'Enter' &&
+        !event.ctrlKey &&
+        !event.metaKey) {
+        let cur = cm.getCursor()
+        let linkStart = cm.getOption('zkn').linkStart
+        let linkEnd = cm.getOption('zkn').linkEnd
+        let line = cm.getLine(cur.line)
+
+        let linkStartPos = line.indexOf(linkStart)
+        let linkEndPos = line.indexOf(linkEnd)
+
+        if (linkStartPos !== -1 &&
+          linkEndPos !== -1 &&
+          linkStartPos < linkEndPos &&
+          (linkStartPos - 1) <= cur.ch &&
+          (linkEndPos + linkEnd.length) >= cur.ch) {
+          let linkName = line.substr(linkStartPos + linkStart.length, linkEndPos - linkStartPos - linkStart.length)
+          if (event.shiftKey) {
+            // Performa a save to avoid a problem where a file
+            // is saved while trying to create or open a link.
+            // That action cancels the file open/create
+            this._renderer.saveFile()
```

Comment:
This will not work as intended, as `saveFile` is a strictly asynchronous event. It sends the file to the main process in order to save, but immediately returns, which might result in unwanted behaviour if the write process of the file is not done while the main process is already instructed to create & open a new file. From the top of my head I don't really know of a way to handle this, so I'll have to come back to this 🧐

Reply:
I see that saveFile can be called synchronously if we added a parameter to saveFile in the renderer that could be used only for cases like this. I'm looking forward to seeing how this code is refactored with the new fsal changes.